### ISSUE TITLE
No gradient flow for PyTorch Soft Rank

### ISSUE THREAD

**Issue by niopeng** [google-research/fast-soft-sort#10]
Hi,

Thanks for the neat package but I have an issue regarding gradient flow using soft rank in PyTorch. Here is how to reproduce it:
<img width="1581" alt="Screen Shot 2021-01-06 at 11 33 04 AM" src="https://user-images.githubusercontent.com/16532408/103822863-bd027900-5025-11eb-827b-9abc747a8a4a.png">
In this example, Should the gradient of x be not all zeros? In comparison, the soft sort seems to work fine:
<img width="1586" alt="Screen Shot 2021-01-06 at 11 33 38 AM" src="https://user-images.githubusercontent.com/16532408/103822901-d86d8400-5025-11eb-88af-a203a691b2b8.png">

In addition, is it possible to add GPU support for soft rank in PyTorch?

Thanks,
Nio

**Comment by niopeng** [comment]
The loss function is not right, sorry about the confusion!
